generators/pkg/generator/generate.go

```diff
@@ -25,3 +45,94 @@ func GenerateTask(spec GitHubSpec) v1beta1.Task {
 	)
 	return task
 }
+
+func addTask(url string) (v1beta1.Task, error) {
+	task := v1beta1.Task{}
+	response, err := http.Get(url)
+	if err != nil {
+		return task, fmt.Errorf("fail to get git-clone task from %s: %w", url, err)
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return task, fmt.Errorf("fail to read reponse body: %v", err)
+	}
+
+	if err := yaml.Unmarshal(body, &task); err != nil {
+		return task, fmt.Errorf("unable to unmarshal git-clone task: %w", err)
+	}
+	task.Labels = map[string]string{"generator.tekton.dev": task.Name}
+	return task, nil
+}
+
+func GeneratePipeline(github GitHub) (v1beta1.Pipeline, []v1beta1.Task, error) {
+	gitCloneTask, err := addTask(GitCloneURL)
+	if err != nil {
+		return v1beta1.Pipeline{}, []v1beta1.Task{}, fmt.Errorf("fail to add git-clone task: %w", err)
+	}
+	buildTask := GenerateTask(github)
+
+	ws := "source"
+	name := "build-from-git-repo"
+	tasksName := []string{"fetch-git-repo", "build-from-repo"}
+
+	pipeline := v1beta1.Pipeline{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "Pipeline",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"generator.tekton.dev": name},
+		},
+		Spec: v1beta1.PipelineSpec{
+			Tasks: []v1beta1.PipelineTask{
+				{
+					Name: tasksName[0],
+					TaskRef: &v1beta1.TaskRef{
+						Name: gitCloneTask.Name,
+					},
+					Params: []v1beta1.Param{
+						{
+							Name: gitCloneTask.Spec.Params[0].Name,
+							Value: v1beta1.ArrayOrString{
+								Type:      v1beta1.ParamTypeString,
+								StringVal: github.Spec.URL,
+							},
+						},
+					},
+					Workspaces: []v1beta1.WorkspacePipelineTaskBinding{
+						{
+							Name:      gitCloneTask.Spec.Workspaces[0].Name,
+							Workspace: ws,
+						},
+					},
+				},
```

Comment:
Isn't this the same as `gitCloneTask`? If not, why is this different, and why is this separated out? (we should make sure to add comments to document this!)

Reply:
This is the same as `gitCloneTask`. Here I refer the `gitCloneTask` and add the parameter value.